### DISCUSSION TITLE
Native concatenation for `TSDataset`

### DISCUSSION THREAD

**Discussion by brsnw250** [etna-team/etna#328] [Improvements]
## Description

Currently, TSDataset can only be split using train_test_split (and only by timestamp). It would be helpful to be able to do the reverse operation. Moreover, it would be convenient to have the possibility to do this by timestamp, segment, or feature.

## Thoughts

We could implement the `TSDataset.concatenate` method with the following signature
```python
def concatenate(
    self,
    other: TSDataset,
    axis: Union[int, str],
    levels: List[Union[int, str]],
    inplace: bool = False
) -> Optional[TSDataset]:
    pass
```
Parameters `axis` and `levels` similar to [`pd.concat`](https://pandas.pydata.org/docs/reference/api/pandas.concat.html#pandas-concat)

## Related

#327 
#310

**Comment by alex-hse-repository** [comment]
May you add some examples where it might be usefull?
